### PR TITLE
Méli-mélo files - Distance calculator and Collection sort

### PR DESCRIPTION
**Pull request by donmcdill**
2023-09-distance-calculator
2023-09-collection-sort

### CODE REVIEW

**Comment by duboisp** [comment]
Pre-approved upon testing and review if it do satisfy to all méli-mélo requirement.

This will be packaged with 2024-09-kejimkujik and probably into 2023-10-mount-revelstoke if kejimkujik is not available on AEM yet.

File: méli-mélo/2023-09-distance-calculator/index.html
Comment (duboisp) [inline méli-mélo/2023-09-distance-calculator/index.html:108]: Are we missing the CSS class `gc-distance-calculator`?

Comment (duboisp) [inline méli-mélo/2023-09-distance-calculator/index.html:108]: I think the data attribute name need to be renamed to match the plugin name

File: méli-mélo/2023-09-distance-calculator/index.html
Comment (duboisp) [inline méli-mélo/2023-09-distance-calculator/index.html:110]: I think this is a Class selector and not a web component (element) selector
```suggestion
		"location":".gc-distance-location",
```

File: méli-mélo/2023-09-distance-calculator/index.html
Comment (duboisp) [inline méli-mélo/2023-09-distance-calculator/index.html:227]: Considering those element is updated and do show status. Adaptation might (more likely) is required for a conformance with the [SC 4.1.3 Status Message](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html)

File: méli-mélo/2023-09-distance-calculator/index.html
Comment (duboisp) [inline méli-mélo/2023-09-distance-calculator/index.html:218]: I did some local test and it was weird to see 3 times the city name.

image

---

Note: Is the text in parenthesis represent the subject type, because Québec city is not the same as Québec the province. Having it rendered like in my screen capture would create confusion.

For a comparaison, here a search but where a province were detected

image

**Review by duboisp — CHANGES_REQUESTED** [review]
A few change requested and I was unable to test collection-sort.

Work fine from a functional after I fixed the little config issue. not a blocker but something to look in a future how duplicate city name should be handle. For example Richmond do exist in multiple provinces. I found there is ~87 duplicate city name.

I found the return object from geogratis do provide all alternative, probably this could be presented to the user somehow.

Considering the data displayed to the use is taken from the API directly and that API is unilingual, we would need also a French example that will show the French equivalent version of that API.

File: méli-mélo/2023-09-distance-calculator/index.html
Comment (donmcdill) [reply méli-mélo/2023-09-distance-calculator/index.html:232]: Added simple status message to example in 9b57565

File: méli-mélo/2023-09-distance-calculator/index.html
Comment (donmcdill) [reply méli-mélo/2023-09-distance-calculator/index.html]: Corrected in commit 9b57565

File: méli-mélo/2023-09-collection-sort/index.html
Comment (donmcdill) [reply méli-mélo/2023-09-collection-sort/index.html:1]: Corrected in 2313541

File: méli-mélo/2023-09-distance-calculator/index.html
Comment (donmcdill) [reply méli-mélo/2023-09-distance-calculator/index.html:223]: Parsing the returned title is something I have been considering. 

It can be more meaningful with something like Lac Gatineau returning "Lac-Gatineau, Mont-Laurier; Antoine-Labelle, Quebec (Unincorporated area)".

The text in the parenthesis appears to be the type of returned location as a whole (Town), (City), (Province), and others. So not Québec city but the returned Gatineau, Quebec is a (City).

File: méli-mélo/2023-09-distance-calculator/js/gc-distance-calculator.js
Comment (donmcdill) [reply méli-mélo/2023-09-distance-calculator/js/gc-distance-calculator.js:1]: I removed the "gc" prefix

**Comment by duboisp** [comment]
I will do a second review and if everything fit according to our méli-mélo checklist, we will merge it.

**Review by duboisp — COMMENTED** [review]
As this was pre-approved last week, I will do another local test + checking that we do comply according to our méli-mélo checklist.

**Review by duboisp — CHANGES_REQUESTED** [review]
I completed the review of hte 2023-09-distance-calculator

There is one HTML parsing error regarding the use of aria-control, I suggest to remove it for now and the configuration of the alternate language link.

I might just push those change in once I completed the review of the collection-sort project so we will be able to merge this PR for this week release.

**Review by duboisp — COMMENTED** [review]
Reviewed and tested locally the 2023-09-collection-sort, everything work.

I did a few comment but not a blocker for now.

File: méli-mélo/2023-09-distance-calculator/index.html
Comment (duboisp) [reply méli-mélo/2023-09-distance-calculator/index.html]: Fixed

File: méli-mélo/2023-09-distance-calculator/index.html
Comment (duboisp) [reply méli-mélo/2023-09-distance-calculator/index.html]: Fixed

File: méli-mélo/2023-09-collection-sort/index.html
Comment (duboisp) [reply méli-mélo/2023-09-collection-sort/index.html:142]: To be looked at in a future version

File: méli-mélo/2023-09-collection-sort/index.html
Comment (duboisp) [reply méli-mélo/2023-09-collection-sort/index.html:276]: To be looked at in a future version

**Review by duboisp — APPROVED** [review]
Change requested was addressed.

I did review and tested locally both méli-mélo project (2023-09-collection-sort & 2023-09-distance-calculator) and they do comply to our méli-mélo minimum requirement.

**Pull request merged**
